src/controllers/controllermanager.cpp

```diff
@@ -37,6 +37,12 @@ const int kPollIntervalMillis = 5;
 const int kPollIntervalMillis = 1;
 #endif
 
+// Note: A standard Midi device runs at 31.25 kbps,
+// a usual Midi message has 30 bits so we may not expect more than
+// 1000 messages per second
+// PortMidiController uses a buffer of 64 messages which results in
+// Maximum polling interval of 64 ms a theoritical minimum of 1 ms
+
```

Comment:
Great, you have such a critical device. On which OS did you test? 

I am little in doubt that a 5 times loop is the best solution for your device. 
It would be really great, if you could provide additional test data. 
* a snipped of --controllerDebug when turning the jog 
* A scoped timer that tracks pollDevices duration.
* Maximum count of messages by one poll.  

There is probably an other big issue in the VCI400 "jog" code. (I am just reading the code) 

If we assume the controller emits a midi message whenever it passes jog degree or an amount of time.
We have 
1  2  3  4  5  6  7  8  9  10  11    
If Mixxx calls poll, it will receive some messages at the first poll, lets say 1234 they are processed accumulated 
and timed. If we got message 5 in the meantime, it is processed in the next inner loop. If we got message 6 in the meantime it is processed in a following loop. So it seams to be luck when the loop is exited, not that nice. 
It becomes worst if this over all loop takes longer than the polling timer, one timer interval is ditched.
Processing midi messages one by one consumes more CPU than required if we process them burst by burst.  
So the Jog sampling is very jittery and unreliable.

It would be better, to use a constant sampling rate and collect ONLY the jogweel samples that have been received since the last timer interval. This way we have fixed timer interval, that will increase the speed stability during scratching.  

I am curious to hear if the test data verify this theory or not.